src/gui/properties/peerlistsortmodel.cpp

```diff
@@ -29,20 +29,21 @@
 #include "peerlistsortmodel.h"
 
 #include "base/utils/string.h"
-#include "peerlistdelegate.h"
+#include "peerlistwidget.h"
 
 PeerListSortModel::PeerListSortModel(QObject *parent)
     : QSortFilterProxyModel(parent)
 {
+    setSortRole(UnderlyingDataRole);
 }
 
 bool PeerListSortModel::lessThan(const QModelIndex &left, const QModelIndex &right) const
 {
     switch (sortColumn()) {
-    case PeerListDelegate::IP:
-    case PeerListDelegate::CLIENT: {
-            const QString strL = left.data().toString();
-            const QString strR = right.data().toString();
+    case PeerListWidget::IP:
```

Comment:
Model should declare it. Don't do it topsy-turvy.

Reply:
but we use StandardModel do you want me to make dummy model?

Reply:
>but we use StandardModel 

I didn't notice there are no compile Model/View design here. Then No, of course, unless u want to implement Model/View completely (applied to all my comments in similar PRS, but please notify where is StandardModel is used as well).

Reply:
> u want to implement Model/View completely

nah I think it's fine as it is,